### PR TITLE
Account dao refactor

### PR DESCRIPTION
**Pull request by alxdarksage**
In this first step, an AccountService is introduced that forwards everything from the existing AccountDao. This changes a bunch of references from the DAO to the service in a mechanical way that should make review easier... the actual refactor will come in a second PR.

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/services/AccountServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    @Test
+    public void getHealthCodeForAccount() {
+        Account account = Account.create();
+        account.setHealthCode(HEALTH_CODE);
+        when(mockAccountDao.getAccount(ACCOUNT_ID)).thenReturn(account);
+        
+        String healthCode = service.getHealthCodeForAccount(ACCOUNT_ID);
+        assertEquals(healthCode, HEALTH_CODE);
+        verify(mockAccountDao).getAccount(ACCOUNT_ID);
+    }
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/services/AccountServiceTest.java:167]: We should also have a unit test for the null case.

File: src/main/java/org/sagebionetworks/bridge/services/AccountService.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+     * For MailChimp, and other external systems, we need a way to get a healthCode for a given email.
+     */
+    public String getHealthCodeForAccount(AccountId accountId) {
+        Account account = getAccount(accountId);
+        if (account != null) {
+            return account.getHealthCode();
+        } else {
+            return null;
+        }
+    }
```
Comment (DwayneJengSage) [inline src/main/java/org/sagebionetworks/bridge/services/AccountService.java:146]: Is it intentional that we're re-implementing getHealthCodeForAccount() instead of calling through to the DAO?

**Review by DwayneJengSage — COMMENTED** [review]
Overall looks good. 2 comments about getHealthCodeForAccount().

File: src/main/java/org/sagebionetworks/bridge/services/AccountService.java (hunk 2 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/services/AccountService.java:146]: No, removed from the DAO.

File: src/test/java/org/sagebionetworks/bridge/services/AccountServiceTest.java (hunk 1 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/services/AccountServiceTest.java:167]: Added

**Review by DwayneJengSage — APPROVED** [review]
